### ISSUE TITLE
GLEW library missing

### ISSUE THREAD

**Issue by antbono** [bhuman/BHumanCodeRelease#48]
Hello everyone!

I am trying to compile the 2021 code release on Pop!_OS 22.04 LTS but it seems that the GLEW library cannot be found by cmake. 
Here the error message

`-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libGL.so   
CMake Error at /snap/clion/198/bin/cmake/linux/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find GLEW (missing: GLEW_INCLUDE_DIRS GLEW_LIBRARIES)
Call Stack (most recent call first):
  /snap/clion/198/bin/cmake/linux/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /snap/clion/198/bin/cmake/linux/share/cmake-3.22/Modules/FindGLEW.cmake:215 (find_package_handle_standard_args)
  /home/toto/robocup/BHumanCodeRelease/Util/SimRobot/Make/Common/OpenGL.cmake:5 (find_package)
  CMakeLists.txt:190 (include)`

I have installed the GLEW library as reported [here](http://glew.sourceforge.net/install.html) in addition to all the packages listed in the [wiki](https://wiki.b-human.de/coderelease2021/getting-started/).

Did anyone have the same problem? Thanks! :)

**Comment by ahasselbring** [comment]
Hi, you can try adding the line
```cmake
set(GLEW_VERBOSE ON)
```
before line 5 (i.e. the `find_package(GLEW)` call) in `Util/SimRobot/Make/Common/OpenGL.cmake`. That should produce some more status output.

By the way, you don't need to follow the instructions at http://glew.sourceforge.net/install.html. Installing `libglew-dev` via apt as indicated by https://wiki.b-human.de/coderelease2021/getting-started/#prerequisites_2 is enough.

**Comment by ahasselbring** [comment]
As a datapoint, I just did the following in a fresh Ubuntu 22.04 docker container:
```bash
apt update
apt install clang cmake git graphviz libasound-dev libglew-dev libqt5opengl5-dev libqt5svg5-dev lld llvm net-tools ninja-build pigz qtbase5-dev rsync wish xterm xxd
git clone --recursive --depth 1 https://github.com/bhuman/BHumanCodeRelease.git
cd BHumanCodeRelease
NO_CLION=true Make/Linux/generate
```
and apart from a problem with pkexec (which isn't relevant for this issue), it runs without errors. Can you try generating the project without CLion?

**Comment by antbono** [comment]
I tried to generate the project without CLion but I received the following messages

-- The CXX compiler identification is Clang 14.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libGL.so   
-- Found GLEW: /usr/include (found version "2.2.0") 
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find ALSA (missing: ALSA_LIBRARY ALSA_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.22/Modules/FindALSA.cmake:62 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:229 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/toto/robocup/BHumanCodeRelease/Build/Linux/CMake/Debug/CMakeFiles/CMakeOutput.log".
-- The CXX compiler identification is Clang 14.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libGL.so   
-- Found GLEW: /usr/include (found version "2.2.0") 
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find ALSA (missing: ALSA_LIBRARY ALSA_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.22/Modules/FindALSA.cmake:62 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:229 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/toto/robocup/BHumanCodeRelease/Build/Linux/CMake/Develop/CMakeFiles/CMakeOutput.log".
-- The CXX compiler identification is Clang 14.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libGL.so   
-- Found GLEW: /usr/include (found version "2.2.0") 
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find ALSA (missing: ALSA_LIBRARY ALSA_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.22/Modules/FindALSA.cmake:62 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:229 (find_package)


-- Configuring incomplete, errors occurred!


I'll try to use a VM. Maybe my system has some missing library

**Comment by ahasselbring** [comment]
ALSA should be part of the package `libasound-dev` (or `libasound2-dev`, I think `libasound-dev` is an alias for `libasound2-dev` in 22.04).

**Comment by antbono** [comment]
Thank you Arne! I did not have libasound-dev installed (and yes it is an alias for libasound2-dev in 22-04). 
Now I can generate the project and compile it. It is also works using CLion. 
I run the command 
`
sudo apt install clang cmake git graphviz libasound-dev libglew-dev libqt5opengl5-dev libqt5svg5-dev lld llvm net-tools ninja-build pigz qtbase5-dev rsync wish xterm xxd`

but I did not notice this problem with the library.

Thank you again for your time and consideration!
